internal/receiver/databricksreceiver/databricks_raw_client.go

```diff
@@ -0,0 +1,95 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package databricksreceiver
+
+import (
+	"fmt"
+	"net/http"
+
+	"go.uber.org/zap"
+
+	"github.com/signalfx/splunk-otel-collector/internal/receiver/databricksreceiver/internal/httpauth"
+)
+
+const (
+	jobsListPath         = "/api/2.1/jobs/list?expand_tasks=true&limit=%d&offset=%d"
+	activeJobRunsPath    = "/api/2.1/jobs/runs/list?active_only=true&limit=%d&offset=%d"
+	completedJobRunsPath = "/api/2.1/jobs/runs/list?completed_only=true&expand_tasks=true&job_id=%d&limit=%d&offset=%d"
+	clustersListPath     = "/api/2.0/clusters/list"
+	pipelinesPath        = "/api/2.0/pipelines"
+	pipelinePath         = "/api/2.0/pipelines/%s"
+)
+
+// databricksRawClient defines methods extracted from databricksRawHTTPClient so
+// that it can be swapped for testing.
+type databricksRawClient interface {
+	jobsList(limit int, offset int) ([]byte, error)
+	activeJobRuns(limit int, offset int) ([]byte, error)
+	completedJobRuns(id int, limit int, offset int) ([]byte, error)
+	clustersList() ([]byte, error)
+	pipelines() ([]byte, error)
+	pipeline(string) ([]byte, error)
+}
+
+// databricksRawHTTPClient wraps an authClient, encapsulates calls to the databricks API, and
+// implements databricksRawClient. Its methods return byte arrays to be unmarshalled
+// by the caller.
+type databricksRawHTTPClient struct {
+	logger     *zap.Logger
+	authClient httpauth.Client
+	endpoint   string
+}
+
+func newDatabricksRawClient(tok, endpoint string, httpClient *http.Client, logger *zap.Logger) databricksRawClient {
+	return &databricksRawHTTPClient{
+		authClient: httpauth.NewClient(httpClient, tok),
+		endpoint:   endpoint,
+		logger:     logger,
+	}
+}
+
+func (c databricksRawHTTPClient) jobsList(limit int, offset int) (out []byte, err error) {
+	path := fmt.Sprintf(jobsListPath, limit, offset)
```

Comment:
Is this the only use of this template string? if so it could help w/ readability to not have it as a const so the verbs are on the same line. Same w/ other constants.

Reply:
Sure. done. (I put them together as consts so you could see what endpoints the class accessed just by looking at the top of the file but maybe that wasn't helpful)